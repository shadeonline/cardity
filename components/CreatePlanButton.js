import React from 'react';
import { View, Text, StyleSheet, Dimensions, TouchableOpacity } from 'react-native';

const CreatePlanButton = ({ text, onPress }) => {
  const screenWidth = Dimensions.get('window').width;
  const cardWidth = screenWidth - 40; // Subtracting padding or margin

  return (
    <TouchableOpacity onPress={onPress} style={[styles.container, { width: cardWidth }]}>
      <Text style={styles.text}>+</Text>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 220,
    backgroundColor: '#e8e9ed',
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 20,
  },
  text: {
    fontSize: 25,
  },
});

export default CreatePlanButton;
