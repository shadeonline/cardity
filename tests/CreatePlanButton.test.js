import React from 'react';
import { render } from 'react-native-testing-library';
import CreatePlanButton from '../components/CreatePlanButton';

describe('CreatePlanButton Component', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(<CreatePlanButton text="Create New Plan" onPress={() => {}} />);
    // Use Jest's expect to match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
