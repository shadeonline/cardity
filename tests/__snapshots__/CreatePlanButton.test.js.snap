// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlanButton Component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e8e9ed",
      "borderRadius": 10,
      "height": 220,
      "justifyContent": "center",
      "marginBottom": 20,
      "opacity": 1,
      "width": 710,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 25,
      }
    }
  >
    +
  </Text>
  <Text
    style={
      {
        "fontSize": 25,
      }
    }
  >
    Create New Plan
  </Text>
</View>
`;
